src/lucky_migrator/create_table_statement.cr

```diff
@@ -73,3 +96,43 @@ class LuckyMigrator::CreateTableStatement
     end
   end
 end
+
+# Encapsulates the building of an index string.
+# Provides an add_to method that accepts a reference to an array and adds the
+# index string or raises an exception if already added.
+#
+# ### Usage
+#
+# ```
+# indices = [] of String
+# IndexDefinition.new('users', column: 'email', using: :btree, unique: true).add_to(indices)
+# ```
+struct IndexDefinition
```

Comment:
I love this extraction ❤️

Reply:
I think it can go in it's own file though since it will probably be reused for other methods I think

Reply:
What do you think about the naming? Should it be `create_index_statement.cr` or something like that for consistency?

Reply:
@mikeeus Yeah that's a good idea CreateIndexStatement in create_index_statement.cr 👍

Reply:
And should it be a class in the LuckyMigrator module? `LuckyMigrator::CreateIndexStatement` or should I leave it as a plain struct?

Reply:
It should be under the `LuckyMigrator` namespace, but it can be a struct `struct LuckyMigrator::CreateIndexStatement`. It could also be a class. Performance isn't really an issue so if the compiler complains it could be a class instead

Reply:
kk great, thanks